torch/distributed/_sharding_spec/api.py

```diff
@@ -1,11 +1,10 @@
 from abc import ABC
 from dataclasses import dataclass
 from typing import List, Union
+from torch.distributed.remote_device import _RemoteDevice
```

Comment:
It looks good to me to have a dedicated type wrapping logic. 

(can be done in followup PRs) Regarding the naming, will it work if we follow existing `torch.device` 's convention? Sth like:

```python
remote_device = torch.distributed.device("worker0/cuda:0")
print(f"{remote_device.worker_name}, {remote_device.rank}, {remote_device.index}")
```

Reply:
I don't have a strong opinion on this. My feeling is that `distributed` might already be sufficient to tell it's not a local device. Please feel free to go with either one. :)